packages/react/src/river/RiverAccordion/RiverAccordion.tsx

```diff
@@ -0,0 +1,254 @@
+          className={clsx(styles.RiverAccordion, styles[`RiverAccordion__align-${align}`], className)}
+          {...rest}
+        >
+          <div className={styles.RiverAccordion__accordionContainer}>{items}</div>
+          <div className={styles.RiverAccordion__visualsContainer}>
+            <div className={styles.RiverAccordion__visualsWrapper}>{visuals}</div>
+          </div>
+        </div>
+      </RiverAccordionContext.Provider>
+    )
+  },
+)
+
+export type RiverAccordionItemProps = React.HTMLAttributes<HTMLDivElement> & {
+  children: React.ReactElement<RiverAccordionHeadingProps | RiverAccordionContentProps>[]
+  index?: number
+}
+
+const RiverAccordionItem = ({className, index, children, ...props}: RiverAccordionItemProps) => {
+  const {openIndex} = useRiverAccordionContext()
+  const panelId = useId()
+  const isOpen = index === openIndex
+
+  const itemContextValue = useMemo(
+    () => ({
+      id: panelId,
+      // Index optional for consumers, but always provided by parent
+      index: index as number,
+      isOpen,
+    }),
+    [panelId, index, isOpen],
+  )
+
+  const {heading, content, visual} = useMemo(() => {
+    return React.Children.toArray(children).reduce<{
+      heading: React.ReactElement<RiverAccordionHeadingProps> | null
+      content: React.ReactElement<RiverAccordionContentProps> | null
+      visual: React.ReactElement<RiverAccordionVisualProps> | null
+    }>(
+      (acc, child) => {
+        if (isRiverAccordionHeading(child)) {
+          acc.heading = child
+        }
+
+        if (isRiverAccordionContent(child)) {
+          acc.content = child
+        }
+
+        if (isRiverAccordionVisual(child)) {
+          acc.visual = child
+        }
+
+        return acc
+      },
+      {heading: null, content: null, visual: null},
+    )
+  }, [children])
+
+  if (!heading || !content || !visual) {
+    return null
+  }
+
+  return (
+    <RiverAccordionItemContext.Provider value={itemContextValue}>
+      <div
+        className={clsx(styles.RiverAccordion__item, isOpen && styles['RiverAccordion__item--open'], className)}
+        {...props}
+      >
+        {heading}
+        <div
+          className={styles.RiverAccordion__panel}
+          id={panelId}
+          aria-hidden={!isOpen}
+          {...(!isOpen && {inert: 'true'})}
+        >
+          {content}
+          {visual}
+        </div>
+      </div>
+    </RiverAccordionItemContext.Provider>
+  )
+}
+
+export type RiverAccordionHeadingProps = HeadingProps
+
+const RiverAccordionHeading = ({className, children, ...props}: RiverAccordionHeadingProps) => {
+  const {setOpenIndex} = useRiverAccordionContext()
+  const {id, index, isOpen} = useRiverAccordionItemContext()
+
+  const onClick = useCallback(() => {
+    if (!isOpen) {
+      setOpenIndex(index)
+    }
+  }, [index, isOpen, setOpenIndex])
+
+  return (
+    <Heading size="6" as="h3" className={clsx(styles.RiverAccordion__heading, className)} {...props}>
```

Comment:
Stylistic nit but the string values for `size` and `as` prop values were throwing me off a little. To make it clearer (and consistent with how we do it in other components) can we please destructure these values in the function arguments and set the default values there? Having the users overrides spread in with `...props` was a less explicit IMO.